### PR TITLE
chore(docs): Solidity Learning Journey

### PR DESCRIPTION
**Pull request by signorecello**
# Description

As part of DevRel's work on Learning Journey's, this PR improves them.

## Problem\*

Relates to [this DevRel issue](https://github.com/AztecProtocol/dev-rel/issues/119)

## Summary\*

- Puts the "Creating a Project" and "Project Breakdown" pages inside a dropdown called "Hello Noir"
- Refactors the Solidity Verifier How-To to include a Remix IDE verification example with images and more context
- Refactors the Noir landing page to incorporate a "Noir Basics" role, improves UI, adds a header image and a tabs feature, and favouring links to awesome-noir instead of hardcoded library examples
- Refactors the Homepage with buttons to immediately start the Solidity Learning journey

## Additional Context

Here's the Solidity Learning Journey. For now only 3 levels deep, assuming the user would have enough autonomy to proceed after the 3rd page. This will be reassessed in the future.

Screenshot 2024-01-18 at 10 29 44

### CODE REVIEW

**Comment by signorecello** [comment]
Could use @jzaki's Solidity knowledge to double-check the guide :)

File: docs/docs/how_to/how-to-solidity-verifier.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+function castVote(bytes calldata proof, uint proposalId, uint vote, bytes32 nullifierHash) public returns (bool) {
+  // ...
+  bytes32[] memory publicInputs = new bytes32[](4);
+  publicInputs[0] = merkleRoot;
+  publicInputs[1] = bytes32(proposalId);
+  publicInputs[2] = bytes32(vote);
+  publicInputs[3] = nullifierHash;
+  require(verifier.verify(proof, publicInputs), "Invalid proof");
+```
+
```
Comment (jzaki) [inline docs/docs/how_to/how-to-solidity-verifier.md:142]: The following three boxes (info, tip, tip), have valuable info, but would benefit from being separate from section `## Step 3 - Deploying`

File: docs/docs/getting_started/hello_noir/index.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Creating A Project
```
Comment (critesjosh) [inline docs/docs/getting_started/hello_noir/index.md:2]: I actually like "Creating A Project" better as a title. It is more clear what the page is about.

File: docs/docs/how_to/how-to-solidity-verifier.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+The order of these inputs would be flattened to: `[x, nested.t1.val1, nested.t1.val2, nested.is_true, y]`
+
+:::
+
+The other function you can call is our entrypoint `verify` function, as defined above.
+
+:::tip
+
+It's worth noticing that the `verify` function is actually a `view` function. This means that its execution does not alter the blockchain state, so it doesn't need to be distributed, and therefore doesn't cost any gas. The EVM node can verify the proof locally.
```
Comment (critesjosh) [inline docs/docs/how_to/how-to-solidity-verifier.md:200]: what does "so it doesn't need to be distributed," mean?

File: docs/docs/how_to/how-to-solidity-verifier.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+For example, chains like `zkSync ERA` and `Polygon zkEVM` do not currently support these precompiles, so proof verification via Solidity verifier contracts won't work. Here's a quick list of EVM chains that have been tested and are known to work:
+
+- Optimism
+- Arbitrum
+- Polygon PoS
+- Scroll
+- Celo
+
+If you test any other chains, please open a PR on this page to update the list. See [this doc](https://github.com/noir-lang/noir-starter/tree/main/with-foundry#testing-on-chain) for more info about testing verifier contracts on different EVM chains.
```
Comment (critesjosh) [inline docs/docs/how_to/how-to-solidity-verifier.md:220]: Do we want to include any other suggested pages for readers at the bottom of the page, other than this? 
image

File: docs/docs/index.mdx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<img src={require("@site/static/img/logo.png").default} style={{display: "block", width: "50%", margin: "2rem auto"}} alt="Noir Logo"  />
+
+Noir is a Domain-Specific Language for SNARK proving systems developed by [Aztec Labs](https://aztec.network/). It allows you to generate complex Zero-Knowledge Programs (ZKP) by using simple and flexible syntax, requiring no previous knowledge on the underlying mathematics or cryptography.
+
+ZKPs are programs that can generate short proofs of a certain statement without revealing some details about it. You can read more about ZKPs [here](https://dev.to/spalladino/a-beginners-intro-to-coding-zero-knowledge-proofs-c56).
```
Comment (critesjosh) [inline docs/docs/index.mdx:29]:
```suggestion
ZK programs are programs that can generate short proofs of a certain statement without revealing some details about it. You can read more about ZKPs [here](https://dev.to/spalladino/a-beginners-intro-to-coding-zero-knowledge-proofs-c56).
```

File: docs/docs/index.mdx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+## What's new about Noir?
+
+Noir works differently from most ZKP languages by taking a two-pronged path. First, it compiles the program to an adaptable intermediate language known as ACIR. From there, depending on a given project's needs, ACIR can be further compiled into an arithmetic circuit for integration with the proving backend.
```
Comment (critesjosh) [inline docs/docs/index.mdx:33]:
```suggestion
Noir works differently from most ZK languages by taking a two-pronged path. First, it compiles the program to an adaptable intermediate language known as ACIR. From there, depending on a given project's needs, ACIR can be further compiled into an arithmetic circuit for integration with the proving backend.
```

File: docs/docs/index.mdx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    Aztec Labs developed NoirJS, an easy interface to generate and verify Noir proofs in a Javascript environment. This allows for Noir to be used in webpages, mobile apps, games, and any other environment supporting JS execution in a standalone manner.
+  </TabItem>
+</Tabs>
+
+
+## Libraries
+
+Noir is meant to be easy to extend by simply importing Noir libraries just like in Rust. 
+The [awesome-noir repo](https://github.com/noir-lang/awesome-noir#libraries) is a collection of libraries developed by the Noir community.
+Writing a new library is easy and makes code be composable and easy to reuse. See the section on [dependencies](noir/modules_packages_crates/dependencies.md) for more information.
```
Comment (critesjosh) [inline docs/docs/index.mdx:67]: Do we want to point readers to various sections at the bottom of this page? We could direct them to Language Concepts, Explainers and Getting Started?

**Review by critesjosh — CHANGES_REQUESTED** [review]

**Comment by critesjosh** [comment]
Did you have trouble updating to Docusaurus 3? I tried to update the Aztec docs but ran into a ton of parsing errors due to the update to MDX v3. Mainly relating to invalid character in names for autogenerated docs. Did you encounter anything like that?

**Comment by signorecello** [comment]
I updated to that version some weeks/months ago actually... And yes MDX was breaking due to some chars not being escaped (like <, {, etc)

Some other minor things as well. I also had a problem with a webpack bug, that should be fixed in docusaurus 3.1.0 so I'll eventually update to that one at least

File: docs/docs/how_to/how-to-solidity-verifier.md (hunk 1 above)
Comment (signorecello) [reply docs/docs/how_to/how-to-solidity-verifier.md:142]: Seems like the best way is to move forward with a `step 4`, I'll do that

File: docs/docs/how_to/how-to-solidity-verifier.md (hunk 3 above)
Comment (signorecello) [reply docs/docs/how_to/how-to-solidity-verifier.md:200]: I explain it later on `The EVM node can verify the proof locally.` but I'll refactor this to be more explanatory

File: docs/docs/how_to/how-to-solidity-verifier.md (hunk 4 above)
Comment (signorecello) [reply docs/docs/how_to/how-to-solidity-verifier.md:220]: Totally, will add some links

**Pull request merged**